cognite/neat/rules/models/information/_rules.py

```diff
@@ -233,6 +265,30 @@ def set_type_for_default(self):
                     ) from None
         return self
 
+    @field_serializer("max_count", when_used="json-unless-none")
+    def serialize_max_count(self, value: int | float | None) -> int | float | None | str:
+        if isinstance(value, float) and math.isinf(value):
+            return None
+        return value
+
+    @field_serializer("reference", when_used="always")
+    def set_reference(self, value: Any, info: SerializationInfo) -> str | None:
+        if isinstance(info.context, dict) and info.context.get("as_reference") is True:
```

Comment:
do you set `as_reference` when you load rules from somewhere?

Reply:
This is not for loading, it is for serialization. added an extra comment